pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -174,13 +185,24 @@ private static String removeOptionsFromSql(String sql) {
     return matcher.replaceAll("");
   }
 
+  private static List<Expression> convertDistinctSelectList(SqlNodeList selectList) {
+    List<Expression> selectExpr = new ArrayList<>();
+    if (!Pql2Compiler.ENABLE_DISTINCT) {
+      throw new SqlCompilationException("Support for DISTINCT is currently disabled in Pinot");
+    }
+    selectExpr.add(convertDistinctAndSelectListToFunctionExpression(selectList));
+    return selectExpr;
+  }
+
   private static List<Expression> convertSelectList(SqlNodeList selectList) {
     List<Expression> selectExpr = new ArrayList<>();
+
     final Iterator<SqlNode> iterator = selectList.iterator();
     while (iterator.hasNext()) {
       final SqlNode next = iterator.next();
       selectExpr.add(toExpression(next));
```

Comment:
Not part of this PR, but we should deduplicate the columns

Reply:
Good catch. For the DISTINCT code via PQL, I deduplicate columns. For calcite compilation path, I can do that in a follow-up since it needs to be done for selection also.